### ISSUE TITLE
LGBMRegressor strange behavior

### ISSUE THREAD

**Issue by aamster** [Microsoft/LightGBM#694]
I have used LGBMRegressor with default settings. The resulting tree doesn't make sense. It has one parent node with split_gain of -1, internal value of 0, and 2 leaves both with the same value. When I remove the 1 feature being used to split on, the resulting tree just has the next most important feature as the one parent node.

I compared with XGBoost, and it results in a sensible tree.
![download 1](https://user-images.githubusercontent.com/5454341/28242944-213f48ea-6988-11e7-8b71-2aa7ce2c2c3e.png)

**Comment by guolinke** [comment]
It is by design.
This tree is a constant tree (thought it has 2 leave, but their outputs are the same.), whose output value is the average of the label. 
For the regression task, boosting from the average can convergence much faster. 
If you don't like this, you can disable it by set ```boost_from_average=false``` .
